types/ember/index.d.ts

```diff
@@ -738,7 +739,7 @@ declare module 'ember' {
             replaceRoute(name: string, ...args: any[]): void;
             transitionToRoute(name: string, ...args: any[]): void;
             model: any;
-            queryParams: string[] | Array<{ [key: string]: { type: string } }>;
+            queryParams: string[] | Array<{ [key: string]: { type: QueryParamTypes } }>;
```

Comment:
It seems like `queryParams` on a Controller can also just be a single string, as well as an array of strings or objects (at least, I found an example of this in the docs).

There is also a `scope` and `as` property on the `queryParams` object, both of type `string`, that might be worth adding now as well in the interest of making the definition more accurate.

Reply:
Done, https://github.com/emberjs/ember.js/blob/4790fb207c13bdb259451bd5236d572950040ca7/packages/ember-routing/lib/system/route.js#L200